packages/blocks/timer/src/app.tsx

```diff
@@ -1,106 +1,293 @@
+        stepLength,
+    );
+
+    if (newDurationInMs !== initialDurationInMs) {
+      applyTimerState({
+        initialDuration: duration.toString(
+          normalizeDurationMinutesAndSeconds(newDurationInMs),
+        ),
+      });
+    }
+  };
+
+  const handleDurationInputChange = (value: duration.Duration) => {
+    const valueInMs = duration.toMilliseconds(value);
+    const acceptedValue =
+      valueInMs > maxInitialDurationInMs
+        ? duration.parse(maxInitialDurationInMs)
+        : valueInMs < minInitialDurationInMs
+        ? duration.parse(minInitialDurationInMs)
+        : value;
+
+    applyTimerState({
+      initialDuration: duration.toString(
+        normalizeDurationMinutesAndSeconds(acceptedValue),
+      ),
+    });
   };
 
   return (
-    <>
-      <LocalizationProvider dateAdapter={AdapterDateFns}>
-        <TimePicker
-          ampm={false}
-          openTo="hours"
-          views={["hours", "minutes", "seconds"]}
-          inputFormat="HH:mm:ss"
-          mask="__:__:__"
-          label="timer"
-          value={localMillis + new Date(0).getTimezoneOffset() * 60000}
-          onChange={(date: Date | null) => {
-            if (date !== null) {
-              update(null, +date - new Date(0).getTimezoneOffset() * 60000);
-            } else {
-              update(null, new Date(0).getTimezoneOffset() * 60000);
-            }
-          }}
-          renderInput={(params) => <TextField {...params} />}
-          disabled={isActive()}
+    <div className="timer-block">
+      <div className="dial">
+        <div className="dial-ring">
+          <div
+            className="dial-ring-completion"
+            style={{ animationDelay: `-${remainingProportion * 100}s` }}
+          />
+        </div>
+        <DurationInput
+          value={
+            timerStatus === "finished"
+              ? parsedInitialDuration
+              : remainingDuration
+          }
+          disabled={timerStatus === "running"}
+          onChange={handleDurationInputChange}
+        />
+        {timerStatus === "running" ? (
+          <button
+            type="button"
+            aria-label="pause"
+            className="big-button big-button_type_pause"
+            onClick={handlePauseClick}
+          >
+            <span className="big-button__icon" />
+          </button>
+        ) : (
+          <button
+            type="button"
+            aria-label="start"
+            className="big-button big-button_type_play"
+            onClick={handlePlayClick}
+          >
+            <span className="big-button__icon" />
+          </button>
+        )}
+      </div>
+      <div className="button-row">
+        <button
+          aria-label="Less time"
+          className="less-time-button"
+          disabled={
+            timerStatus === "running" ||
+            timerStatus === "paused" ||
+            initialDurationInMs <= minInitialDurationInMs
+          }
```

Comment:
It feels like it would be helpful for users to be able to click the more/less time buttons either (a) at any point, or (b) when the timer is paused. The current behavior – it is only possible to click more/less when the timer has not been started at all – feels overly restrictive.

This could be in follow up. @MaggieAppleton any views?

Reply:
I've just realised that we already allow text input on pause. Changing digits resets the ring. So I've enabled + - on pause in https://github.com/hashintel/hash/pull/576/commits/3836fa57c91c9898377be850aea9e4aa73ba15cf. We can revert if it's not a good guess.

Reply:
Makes sense! Leaving this comment thread unresolved in case Maggie wants to add views.